cpp/dolfinx/mesh/MeshTags.h

```diff
@@ -155,202 +155,229 @@ class MeshTags
 };
 
 /// Create MeshTags from arrays
-/// @param[in] comm
-/// @param[in] mesh
-/// @param[in] entity_cell_type Cell type of entities which MeshTags are
-///   tagging
-/// @param[in] topology Array describing topology of tagged mesh
-///   entities. This array must be using input_global_indices from the
-///   mesh.
-/// @param[in] values Array of values to attach to mesh entities
+/// @param[in] comm The MPI communicator
+/// @param[in] mesh The Mesh that the tags are associated with
+/// @param[in] tag_cell_type Cell type of entities which are being
+///   tagged
+/// @param[in] entities 'Node' indices (using the of entities
+///   input_global_indices from the Mesh) fir vertices of for each
+///   entity that is tagged. The numbers of rows is equal to the number
+///   of entities.
+/// @param[in] values Tag values for each entity in @ entities. The
+///   length of @ values  must be equal to number of rows in @ entities.
 template <typename T>
 mesh::MeshTags<T>
 create_meshtags(MPI_Comm comm, const std::shared_ptr<const mesh::Mesh>& mesh,
-                const mesh::CellType& entity_cell_type,
+                const mesh::CellType& tag_cell_type,
                 const Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic,
-                                   Eigen::RowMajor>& topology,
+                                   Eigen::RowMajor>& entities,
                 const std::vector<T>& values)
 {
-  if ((std::size_t)topology.rows() != values.size())
+  // NOTE: Not yet working for higher-order geometries
+  //
+  // TODO: Avoid expensive-to-create std::vector<std::vector>>. Build
+  //       AdjacencyList instead.
+
+  assert(mesh);
+  if ((std::size_t)entities.rows() != values.size())
     throw std::runtime_error("Number of entities and values must match");
 
-  // Build array of topology node indices
-  std::vector<std::int64_t> topo_unique(
-      topology.data(), topology.data() + topology.rows() * topology.cols());
-  std::sort(topo_unique.begin(), topo_unique.end());
-  topo_unique.erase(std::unique(topo_unique.begin(), topo_unique.end()),
-                    topo_unique.end());
+  // Build sorted array of entity node indices
```

Comment:
I do not see where this `nodes` is used now, is it?

Reply:
Indeed, not longer used. Removed.